### PR TITLE
Support multiple feeds as input to the baseline generator

### PR DESCRIPTION
**Pull request by mmitche**
For -o usage, walk each feed and find a package for download on that feed
For -u usage, choose the newest version among all best matches on all feeds

### CODE REVIEW

**Review by dougbu — COMMENTED** [review]
Looks good but I'm not sure it's useful in 5.0 outside our servicing readiness exercise. Are there servicing for realz scenarios where we'd need to prepare the next baseline prior to a release❔

**Review by wtgodbe — APPROVED** [review]
Port to dotnet/extensions as well?

**Comment by dougbu** [comment]
> Port to dotnet/extensions as well?

If we have use cases, taking it to dotnet/extensions 2.1 and 3.1 would make sense. (I haven't looked at what the new owners are doing in 5.0 and 6.0 branches in that repo.) As I mentioned above, I'm not sure anything but dougbu/5.0.2 really needs this because everything ends up on NuGet.org before we generate the next baselines.

File: eng/tools/BaselineGenerator/Program.cs [hunk 1]
```diff
@@ -30,16 +31,18 @@ static void Main(string[] args)
...
+        private static readonly string[] _defaultSources = new string[] { "https://api.nuget.org/v3/index.json" };
+
         public Program()
         {
-            _source = Option(
-                "-s|--package-source <SOURCE>",
-                "The NuGet source of packages to fetch",
-                CommandOptionType.SingleValue);
+            _sources = Option(
+                "-s|--package-sources <SOURCE>...",
```
Comment (dougbu) [inline eng/tools/BaselineGenerator/Program.cs:43]: My bad, this version of `CommandLineUtils` doesn't support the `...` indicator that more than one is allowed.
```suggestion
                "-s|--package-sources <Sources>",
```

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
Got it working (finally). But, I can't rebase on 'master' with my current setup (yet).

**Comment by mmitche** [comment]
Rebased on master.

**Comment by dougbu** [comment]
I don't see a reason to retry the failing test. Nothing other than the Code Check job could possibly be affected by this change. Merge away&hellip;

**Comment by dougbu** [comment]
Do you want me to merge this @mmitche❔

**Pull request merged**